File: clojure-repo-service/src/org/cyverse/dataone/ClojureRepoServiceFactory.clj
(ns org.cyverse.dataone.ClojureRepoServiceFactory
  (:gen-class :extends org.irods.jargon.dataone.reposervice.AbstractDataOneRepoFactory))

(defn -instance [this publication-context irods-account]
  (org.cyverse.dataone.ClojureRepoService. irods-account publication-context))

File: clojure-repo-service/src/java/org/cyverse/dataone/ClojureRepoServiceFactory.java
package org.cyverse.dataone;

import org.irods.jargon.core.connection.IRODSAccount;
import org.irods.jargon.dataone.reposervice.AbstractDataOneRepoServiceFactory;
import org.irods.jargon.dataone.reposervice.AbstractDataOneRepoServiceAO;
import org.irods.jargon.dataone.plugin.PublicationContext;

public class ClojureRepoServiceFactory extends AbstractDataOneRepoServiceFactory {

    public AbstractDataOneRepoServiceAO instance(PublicationContext publicationContext, IRODSAccount irodsAccount) {
        return new ClojureRepoService(irodsAccount, publicationContext);
    }
}
